frame/undercollateralized-loans/src/lib.rs

```diff
@@ -0,0 +1,376 @@
+			LiquidationStrategyId = Self::LiquidationStrategyId,
+		>;
+
+		type Counter: AddAssign
+			+ One
+			+ FullCodec
+			+ Copy
+			+ PartialEq
+			+ PartialOrd
+			+ Debug
+			+ Default
+			+ TypeInfo;
+
+		type UnixTime: UnixTime;
+		type MaxMarketsCounterValue: Get<Self::Counter>;
+		type MaxLoansPerMarketCounterValue: Get<Self::Counter>;
+		// Each payments schedule can not have more than this amount of payments.
+		type MaxPaymentsAmountValue: Get<u32>;
+		type OracleMarketCreationStake: Get<Self::Balance>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	pub type MarketsCounterStorage<T: Config> = StorageValue<_, T::Counter, ValueQuery>;
+
+	#[pallet::storage]
+	pub type LoansCounterStorage<T: Config> = StorageValue<_, T::Counter, ValueQuery>;
+
+	// TODO: @mikolaichuk: implement checking of these counters exceeding some max value.
+	#[pallet::storage]
+	pub type LoansPerMarketCounterStorage<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, T::Counter, ValueQuery>;
+
+	#[pallet::storage]
+	pub type CurrentDateStorage<T: Config> = StorageValue<_, Timestamp, ValueQuery>;
+
+	// Markets storage. AccountId is id of market's account.
+	#[pallet::storage]
+	pub type MarketsStorage<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, MarketInfoOf<T>, OptionQuery>;
+
+	// Loans storage. AccountId is id of loan's account.
+	#[pallet::storage]
+	pub type LoansStorage<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, LoanConfigOf<T>, OptionQuery>;
+
+	// Payments schedule. Keeps sets of loans which have to be paid before the particular block
+	// number.
+	#[pallet::storage]
+	pub type PaymentsScheduleStorage<T: Config> =
+		StorageMap<_, Twox64Concat, T::BlockNumber, BTreeSet<T::AccountId>, ValueQuery>;
+
+	// Use hashmap as a set.
+	#[pallet::storage]
+	pub type NonActiveLoansStorage<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, (), OptionQuery>;
+
+	// Maps market's account id to market's debt token
+	#[pallet::storage]
+	pub type DebtTokenForMarketStorage<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		T::AccountId,
+		<T as DeFiComposableConfig>::MayBeAssetId,
+		OptionQuery,
+	>;
+
+	// Payments schedule storage.
+	// Maps payment moment and loan account id to interest rate for this payment.
+	#[pallet::storage]
+	pub type ScheduleStorage<T: Config> =
+		StorageMap<_, Twox64Concat, Timestamp, BTreeSet<T::AccountId>, ValueQuery>;
+
+	// TODO: @mikolaichuk: storages for borrowers' strikes (local for paricular market and global
+	// for all markets).
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		MarketCreated { market_info: MarketInfoOf<T> },
+		LoanCreated { loan_config: LoanConfigOf<T> },
+		LoanContractWasExecuted { loan_config: LoanConfigOf<T> },
+		LoanWasNotFoundInTeStorage { loan_account_id: T::AccountId },
```

Comment:
NIT: "LoanWasNotFoundInTeStorage" -> "LoanWasNotFoundInStorage" or "LoanNotInStorage"

Reply:
Removed this event since it is not used.